src/e2ee.rs

```diff
@@ -80,228 +76,221 @@ impl E2eeHelper {
             Err(err) => {
                 bail!("Failed to load own public key: {}", err);
-            } 
-            Ok(public_key) => public_key
+            }
+            Ok(public_key) => public_key,
         };
 
-        /* libEtPan's pgp_encrypt_mime() takes the parent as the new root. We just expect the root as being given to this function. */
-        let prefer_encrypt = if 0
-            != context
-                .sql
-                .get_config_int(context, "e2ee_enabled")
-                .unwrap_or_default()
-        {
+        let e2ee = context.sql.get_config_int(&context, "e2ee_enabled");
+
+        let prefer_encrypt = if 0 != e2ee.unwrap_or_default() {
             EncryptPreference::Mutual
         } else {
             EncryptPreference::NoPreference
         };
 
+        let mut encryption_successfull = false;
+        let mut do_encrypt = false;
+        let mut keyring = Keyring::default();
+        let mut peerstates: Vec<Peerstate> = Vec::new();
 
-        let plain: *mut MMAPString = mmap_string_new(b"\x00" as *const u8 as *const libc::c_char);
-            /*only for random-seed*/
-            if prefer_encrypt == EncryptPreference::Mutual || e2ee_guaranteed {
-                do_encrypt = true;
-                for recipient_addr in recipients_addr.iter() {
-                    if *recipient_addr != addr {
-                        let peerstate =
-                            Peerstate::from_addr(context, &context.sql, &recipient_addr);
-                        if peerstate.is_some()
-                            && (peerstate.as_ref().unwrap().prefer_encrypt
-                                == EncryptPreference::Mutual
-                                || e2ee_guaranteed)
-                        {
-                            let peerstate = peerstate.unwrap();
-                            info!(context, "dc_e2ee_encrypt {} has peerstate", recipient_addr);
-                            if let Some(key) = peerstate.peek_key(min_verified as usize) {
-                                keyring.add_owned(key.clone());
-                                peerstates.push(peerstate);
-                            }
-                        } else {
-                            info!(
-                                context,
-                                "dc_e2ee_encrypt {} HAS NO peerstate {}",
-                                recipient_addr,
-                                peerstate.is_some()
-                            );
-                            do_encrypt = false;
-                            /* if we cannot encrypt to a single recipient, we cannot encrypt the message at all */
-                            break;
+        /*only for random-seed*/
+        if prefer_encrypt == EncryptPreference::Mutual || e2ee_guaranteed {
+            do_encrypt = true;
+            for recipient_addr in recipients_addr.iter() {
+                if *recipient_addr != addr {
```

Comment:
it is slightly better to do `if recipient_addr != &addr` to avoid the copy/deref usually.